### ISSUE TITLE
API to update SSL certificates programmatically 

### ISSUE THREAD

**Issue by theDepart3d** [NginxProxyManager/nginx-proxy-manager#2855]
**Is your feature request related to a problem? Please describe.**
I have this frustration every 3 months where I have to manually update 70% of my sub domains of all my certificates manually as it gets issued on my main server. 

My NPM Setup:

1.  WAN -> dedicated server
2. Dedicated Server -> reverse proxy (nginx for some subdomains) -> NPM for locally hosted sub domains -> local server

All certificates are issued and used on the dedicated server. So I basically have to manually scp around 14 domains ssl certificates to my desktop and then manually updated the ssl certificate using NPM dashboard.

**Describe the solution you'd like**
The ability to automatically add/update SSL certificates for domains and sub-domains. This could also work with adding/updating/removing domains and sub-domains from NPM

**Describe alternatives you've considered**
Programmatically use bash to update  the certificate and key. Which is a bit hard as the `custom_ssl` folder does not follow any naming conventions. It has `/custom_ssl/npm-56` etc, Which is quite hard to determine what key and cert belongs to what domain.

**Additional context**
**EG:** 
 ![custom_ssl_npm](https://user-images.githubusercontent.com/109445905/234073390-14e3d2d6-b184-4546-b98e-8ab38100d0f2.png)

**Comment by ysmintor** [comment]
@theDepart3d I agree with this problem. I just use the only one wildcard certificate for the domain and subdomain. And manually replace the old certificate and then reboot NPM docker.

NPM now only provide upload method which is not enough for us to do auto update ssl certificate. I also have your problem. So I hope author can provide method to auto replace certificate and reload NPM with the new certificate.

**Comment by ysmintor** [comment]
I hope someone can provide a better way to update certificate. I use one VPS via ACME.sh to get the Wildcard certificates and then sync to slave master to use. Currently, I restart NPM everyday to update the new file. But I would like a more elegant way to notify NPM the new certificates updated.

**Comment by theDepart3d** [comment]
> I hope someone can provide a better way to update certificate. I use one VPS via ACME.sh to get the Wildcard certificates and then sync to slave master to use. Currently, I restart NPM everyday to update the new file. But I would like a more elegant way to notify NPM the new certificates updated.

I kind of have a solution that might work. Im going to give it a try this weekend. 

What i have noticed is that there is a correlation between the npm-* and the database entries.

I think i may have a method to automate this without actually touching the core NPM. It would simply be an extra shell script. Ill get back to you if there are any successes.

**Comment by ysmintor** [comment]
> > I hope someone can provide a better way to update certificate. I use one VPS via ACME.sh to get the Wildcard certificates and then sync to slave master to use. Currently, I restart NPM everyday to update the new file. But I would like a more elegant way to notify NPM the new certificates updated.
> 
> I kind of have a solution that might work. Im going to give it a try this weekend.
> 
> What i have noticed is that there is a correlation between the npm-* and the database entries.
> 
> I think i may have a method to automate this without actually touching the core NPM. It would simply be an extra shell script. Ill get back to you if there are any successes.

Yes, I actually use the correlation. I create one and then make a soft link to the real wildcard certificates. Since NPM will load it to memory, it will not update the new version once certificates get updated. So I make a shell script to restart NPM docker per day. I hope the author can develop some method to adapt hot update certificate with external certificates. Hope you can success.
